### PR TITLE
David Soto - auth i

### PR DESCRIPTION
**Pull request by David247365**

### CODE REVIEW

**Comment by David247365** [comment]
@CookieMonsta89

File: index.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+const bcrypt = require("bcryptjs");
+const session = require("express-session");
+const KnexSessionStore = require("connect-session-knex")(session);
+
+const db = require("./database/dbConfig.js");
+
+const server = express();
+
+// Remeber that this is a secret
+const sessionConfig = {
```
Comment (CookieMonsta89) [inline index.js]: Session config looks good.

File: index.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    .catch(err =>
+      res.status(500).json({ message: "There was an error, please try again" })
+    );
+});
+
+// Get users
+
+// Protect this route and only allow authenticated users to see it
+server.get("/api/users", (req, res) => {
+  if (req.session && req.session.userId) {
```
Comment (CookieMonsta89) [inline index.js]: You can create a function that does this for you.  ```function protected(req, res, next) {
    if (req.session && req.session.username) {
        next();
    } else {
        return res.status(401).json({error:'Incorrect Credentials'});
    }
}```

Comment (CookieMonsta89) [reply index.js]: Makes it reusable

**Pull request closed without merging**